Sources/TuistGenerator/Mappers/SynthesizedResourceInterfaceProjectMapper.swift

```diff
@@ -193,25 +174,32 @@ public final class SynthesizedResourceInterfaceProjectMapper: ProjectMapping { /
         path: AbsolutePath,
         target: Target
     ) -> String {
-        switch resourceSynthesizer.parser {
-        case .assets, .strings, .fonts, .interfaceBuilder:
-            return target.name.camelized.uppercasingFirst
-        case .plists, .json, .yaml, .coreData:
-            return path.basenameWithoutExt.camelized.uppercasingFirst
-        }
+        return target.name.camelized.uppercasingFirst
     }
 
     private func paths(
         for resourceSynthesizer: ResourceSynthesizer,
-        target: Target
+        target: Target,
+        developmentRegion: String?
     ) -> [AbsolutePath] {
         let resourcesPaths = target.resources
             .map(\.path)
 
         var seen: Set<String> = []
-        return resourcesPaths
+        var paths = resourcesPaths
             .filter { $0.extension.map(resourceSynthesizer.extensions.contains) ?? false }
             .filter { seen.insert($0.basename).inserted }
+            .sorted()
+
+        if resourceSynthesizer.isLocalizable {
```

Comment:
I'd honestly remove the variable `isLocalizable` altogether and do the switch directly. It will be less verbose and more clear on what types we operate on.